openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/DataInsightsApp.java

```diff
@@ -1,17 +1,112 @@
 package org.openmetadata.service.apps.bundles.insights;
 
+import static org.openmetadata.service.exception.CatalogExceptionMessage.INVALID_APP_TYPE;
+
+import com.cronutils.model.Cron;
+import java.util.List;
 import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.api.services.ingestionPipelines.CreateIngestionPipeline;
 import org.openmetadata.schema.entity.app.App;
+import org.openmetadata.schema.entity.app.AppType;
+import org.openmetadata.schema.entity.app.ScheduleType;
+import org.openmetadata.schema.entity.services.ServiceType;
+import org.openmetadata.schema.entity.services.ingestionPipelines.AirflowConfig;
+import org.openmetadata.schema.entity.services.ingestionPipelines.IngestionPipeline;
+import org.openmetadata.schema.entity.services.ingestionPipelines.PipelineType;
+import org.openmetadata.schema.metadataIngestion.DataInsightPipeline;
+import org.openmetadata.schema.metadataIngestion.SourceConfig;
+import org.openmetadata.schema.type.EntityReference;
+import org.openmetadata.schema.type.ProviderType;
+import org.openmetadata.schema.type.Relationship;
+import org.openmetadata.service.Entity;
 import org.openmetadata.service.apps.AbstractNativeApplication;
+import org.openmetadata.service.exception.EntityNotFoundException;
 import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.EntityRepository;
+import org.openmetadata.service.jdbi3.IngestionPipelineRepository;
 import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.util.FullyQualifiedName;
 
 @Slf4j
 public class DataInsightsApp extends AbstractNativeApplication {
 
+  private static final String INGESTION_PIPELINE_NAME = "OpenMetadata_dataInsight";
+  private static final String SERVICE_NAME = "OpenMetadata";
+  private static final String SERVICE_TYPE = "Metadata";
+  private static final String PIPELINE_DESCRIPTION = "OpenMetadata DataInsight Pipeline";
+
   @Override
   public void init(App app, CollectionDAO dao, SearchRepository searchRepository) {
     super.init(app, dao, searchRepository);
     LOG.info("Data Insights App is initialized");
   }
+
+  @Override
+  public void initializeExternalApp() {
+    if (getApp().getAppType() == AppType.External && getApp().getScheduleType().equals(ScheduleType.Scheduled)) {
```

Comment:
isn't this same for all the external apps?
1. Lets centralize this out of the initialize method
2. Create clean abstraction of applicaiton level initialization() which is not related to creating the app and deploying the pipeline

Reply:
created unique `install` method that handles the logic for both internal/external apps. We now have a single init and a single install method to take care of